src/app/components/client/toolbar/AppPicker.tsx

```diff
@@ -0,0 +1,373 @@
+import { useL10n } from "../../../hooks/l10n";
+import { BentoIcon } from "../../server/Icons";
+import { gaEvent } from "../../../functions/client/gaEvent";
+
+const getProducts = (referringHost: string) => ({
+  relay: {
+    id: "relay",
+    url: `https://relay.firefox.com/?utm_source=${encodeURIComponent(
+      referringHost
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+    gaLabel: "fx-relay",
+  },
+  pocket: {
+    id: "pocket",
+    url: "https://app.adjust.com/hr2n0yz?engagement_type=fallback_click&fallback=https%3A%2F%2Fgetpocket.com%2Ffirefox_learnmore%3Fsrc%3Dff_bento&fallback_lp=https%3A%2F%2Fapps.apple.com%2Fapp%2Fpocket-save-read-grow%2Fid309601447",
+    gaLabel: "pocket",
+  },
+  fxDesktop: {
+    id: "fxDesktop",
+    url: `https://www.mozilla.org/firefox/new/?utm_source=${encodeURIComponent(
+      referringHost
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+    gaLabel: "fx-desktop",
+  },
+  fxMobile: {
+    id: "fxMobile",
+    url: `https://www.mozilla.org/firefox/browsers/mobile/?utm_source=${encodeURIComponent(
+      referringHost
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+    gaLabel: "fx-mobile",
+  },
+  vpn: {
+    id: "vpn",
+    url: `https://www.mozilla.org/products/vpn/?utm_source=${encodeURIComponent(
+      referringHost
+    )}&utm_medium=referral&utm_campaign=bento&utm_content=desktop`,
+    gaLabel: "vpn",
+  },
+});
+
+/**
+ * Menu that can be opened to see other relevant products Mozilla has available for people.
+ */
+export const AppPicker = () => {
+  const l10n = useL10n();
+
+  const referringHost =
+    typeof document !== "undefined"
+      ? document.location.host
+      : "monitor.firefox.com";
+  const products = getProducts(referringHost);
+  const linkRefs: Record<
+    keyof typeof products,
+    RefObject<HTMLAnchorElement>
+  > = {
+    relay: useRef<HTMLAnchorElement>(null),
+    pocket: useRef<HTMLAnchorElement>(null),
+    fxDesktop: useRef<HTMLAnchorElement>(null),
+    fxMobile: useRef<HTMLAnchorElement>(null),
+    vpn: useRef<HTMLAnchorElement>(null),
+  };
+  const mozillaLinkRef = useRef<HTMLAnchorElement>(null);
+
+  const onSelect = (itemKey: Key) => {
+    Object.entries(products).forEach(([key, productData]) => {
+      if (itemKey === productData.id) {
+        linkRefs[key as keyof typeof products].current?.click();
+        gaEvent({
+          category: "bento",
+          action: "bento-app-link-click",
+          label: productData.gaLabel,
+        });
+      }
+    });
+    if (itemKey === "mozilla") {
+      mozillaLinkRef.current?.click();
+      gaEvent({
+        category: "bento",
+        action: "bento-app-link-click",
+        label: "Mozilla",
+      });
+    }
+  };
+
+  return (
+    <AppPickerTrigger
+      label={l10n.getString("toolbar-app-picker-trigger-title")}
+      onAction={onSelect}
+      referringHost={referringHost}
+    >
+      <Item
```

Comment:
Suggestion: Could we add the l10n string to `getProducts` and map over the products in order to render the `Item` components? Editing product data would only need to be changed in one place and the JSX would shrink a bit.

Reply:
Good idea: 5dbc6927b